apis/python/tests/test_sparse_nd_array.py

```diff
@@ -388,3 +395,306 @@ def test_sparse_nd_array_reshape(tmp_path):
     a.create(type=pa.int32(), shape=(10, 10, 10))
     with pytest.raises(NotImplementedError):
         assert a.reshape((100, 10, 1))
+
+
+@pytest.mark.parametrize(
```

Comment:
A small nit - in the interest of making this easier to maintain.  Unrolled parameter lists like this are quite challenging to review which cases are covered. It would be easier to read if the options were constructed as a series of loops, eg,

```
for write_format in ['coo']:
    for read_format in ['table', 'csr']:
        for shape in [(4,), (4,6,)]:
            ...
```

I realize that not of the conditions can be created this way.  But even subsets done this way are far more obvious, and can then be concatenated.

As example of why it matters: the tests seem to be missing a passing (non-error case) for `read_format=csr` or `csc`.  

Code coverage would also catch this, but it would be nice if it was easier to visually inspect.